inc/Utf8/Sort.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace dokuwiki\Utf8;
+
+/**
+ * DokuWiki sort functions
+ *
+ * When "intl" extension is available, all sorts are done using a collator.
+ * Otherwise, primitive PHP functions are called.
+ *
+ * The collator is created using the locale given in $conf['lang'].
+ * It always uses case insensitive "natural" ordering in its collation.
+ * The fallback solution uses the primitive PHP functions that return almost the same results
+ * when the input is text with only [A-Za-z0-9] characters.
+ *
+ * @license    GPL 2 (http://www.gnu.org/licenses/gpl.html)
+ * @author     Moisés Braga Ribeiro <moisesbr@gmail.com>
+ */
+class Sort
+{
+    /** @var \Collator[] language specific collators, usually only one */
+    protected static $collator;
+
+    /**
+     * Initialization of a collator using $conf['lang'] as the locale.
+     * The initialization is done only once, except when $reload is set to true.
+     * The collation takes "natural ordering" into account, that is, "page 2" is before "page 10".
+     *
+     * @param bool $reload Usually false; true forces collator re-creation
+     * @return \Collator Returns a configured collator or null if the collator cannot be created.
+     *
+     * @author Moisés Braga Ribeiro <moisesbr@gmail.com>
+     */
+    protected static function getCollator($reload = false)
```

Comment:
$reload is no longer necessary, as all collators get collected in an array.

Reply:
I usually keep a reload mechanism for static classes as it makes testing easier. But I guess you're right - we don't need for this class.